benchmarks/matmulCpp_benchmark/benchmark_info.json

```diff
@@ -0,0 +1,7 @@
+{
+    "name" : "matrix multiplication benchmark in C++",
+    "description" : "Some test of matrices operations in C++",
+    "implementation_file" : "implementation.py",
+    "class_name" : "MatmulBenchmarkCpp",
+    "default_settings" : "settings1.json"
```

Comment:
now that you renamed the jso, settings1.json does not exist any more

Reply:
@gabrielefronze this is why we need to test also the benchmarks. This PR does not break O4BCBenchmark but breaks the benchmark itself. We may need to think how to split the tests in order to run them only on the benchmark that is being modified by the PR, but I think we need automatic testing of all code that is being merged. I would ask @AlessioBorriero to add also in CI.yml a test of the c++ benchmark which should also test it with default settings (i.r. run the cli command without providing a setting file). I will open an issue regarding rethinking CI tests once we have the new implementation

Reply:
I see your point!
The problem here is that the benchmarks are themselves part of the repo, instead of living on another repo (together or one by one).
Philosophically speaking there is no reason to put benchmarks CI in a repo which just handles them and whose base code is not affecting the benchmarks' functionality, nor is the cause of the break (e.g. in this case we have basically no way to fix without modifying the code).
We agree on the fact a rework is needed and will happen since is in the pipeline.
However, I wouldn't spend time in developing the tests here (or at least I wouldn't add them to this repo CI) and wait to add them when the transition is completed.